src/service/OAuthService.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response, Router } from "express"
+import { OAuthConfiguration } from "../utils/OAuthConfiguration"
+import { _verify } from "./Security"
+import fetch, { Request as Req, Response as Res } from "node-fetch"
+import { CredentialService } from './CredentialService';
+import { decode } from "jsonwebtoken"
+
+export class OAuthService {
+  public static _name = "OAuth"
+  public static Router = Router()
+}
+
+OAuthService.Router.get("/oauth/start", async (req: Request, res: Response) => {
```

Comment:
As we discussed, it'd be cleaner if we could remove the need for `/oauth/*` methods. I think an approach that would work the same way is returning `Link` headers in a `HTTP 403 Forbidden` response:
1. Client calls `GET /participant/me` with or without `Authorization` header.
2. Server checks `Authorization` header: 
    1. Server finds `Authorization` header and approves request as currently implemented. 
    2. Server does not find the header and responds `403`. 
        1. **This is where we should add a Link header to indicate OAuth2 login.**

Additionally, we should use a `.well-known` URL to provide this information ahead of time **only if IdP support is enabled.**
 - For example, [OIDC has this connect endpoint](https://swagger.io/docs/specification/authentication/openid-connect-discovery/).
 - We could use the same name or a different name and provide this info there.

Reply:
So I started working on a solution for this and then ran into a couple of different issues that made me realize that while this is a good idea, it would be better if we worked on a couple of things before we attempted this.

1. The URL we would return as a part of the Link Header is incomplete, because the code challenge/code verifier pair are handled in the front end right now. The return URL would not be useable as it is. The solution for this problem would be to move the code challenge/verifier handling into the backend (Even better for security!) and design a way to cache/validate pairs with Redis.
2. Responses for each endpoint are handled separately. That is, each entry point has a try-catch block and in the block we generate the response. Adding a link to each response would need to be done individually on each endpoint.
As a way to make this easier to manage we should include an error handling middleware for Express to centralize the responses and make it easier to manage from a developing point of view. (Also less error prone. We could "forget" to add the link on any endpoint and run into unwanted behaviors)

Reply:
Got it, perhaps we merge this as is for now and track this in the GitHub issue as a v1.1 thing? Are there any other major architectural modifications we anticipate here?